src/agent/src/message_queue_utils.hpp

```diff
@@ -15,3 +17,9 @@ boost::asio::awaitable<std::string> GetMessagesFromQueue(std::shared_ptr<IMultiT
 void PopMessagesFromQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue, MessageType messageType);
 
 void PushCommandsToQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue, const std::string& commands);
+
+std::optional<command_store::Command> GetCommandFromQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue);
```

Comment:
Why don't you return an `awaitable` instead of an `optional`?

Reply:
The function is non-blocking since it calls co_await on its internal timer. Let's keep an eye on this and see if it's necessary, we can change it down the road.